### PR TITLE
Feature/support es256 k

### PR DESCRIPTION
**Pull request by aleksandra-bel**
This is implementation of Support ES256K/secp256k1
Closes: https://github.com/eclipse-tractusx/managed-identity-wallet/issues/266

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [ ] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [ ] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

**Review by borisrizov-zf — CHANGES_REQUESTED** [review]

File: src/main/java/org/eclipse/tractusx/managedidentitywallets/service/JwtPresentationES256KService.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+        try {
+            JWSSigner signer = new ECDSASigner(ecPrivateKey);
+            signer.getJCAContext().setProvider(BouncyCastleProviderSingleton.getInstance());
+            if (!signer.supportedJWSAlgorithms().contains(JWSAlgorithm.ES256K)) {
+                throw new UnsupportedAlgorithmException(String.format("Invalid signing method. Supported signing methods: %s",
+                        signer.supportedJWSAlgorithms().stream().map(Algorithm::getName).collect(Collectors.joining(", "))));
+            } else {
+                JWSAlgorithm algorithm = JWSAlgorithm.ES256K;
+                JOSEObjectType type = JOSEObjectType.JWT;
+                JWSHeader header = new JWSHeader(algorithm, type, null, null, null, null, null, null, null, null, issuer, true, null, null);
```
[the remark this answers is not in the document]
Comment (aleksandra-bel) [reply src/main/java/org/eclipse/tractusx/managedidentitywallets/service/JwtPresentationES256KService.java]: With the builder it looks much better, fixed [541a524](https://github.com/eclipse-tractusx/managed-identity-wallet/pull/274/commits/541a524646b26910d19648f800931d54d549146e)

**Review by borisrizov-zf — APPROVED** [review]
Looks good, will merge after manual tests pass.

**Comment by borisrizov-zf** [comment]
@aleksandra-bel @andreibogus 
Ok, I found something. We have to add the new key to the DID document.

**Review by borisrizov-zf — CHANGES_REQUESTED** [review]
Update DID document, adding the newly issued key to the list of verificationMethod.

**Comment by andreibogus** [comment]
added WalletKey for ES256K algorithm to DidDocument

**Review by borisrizov-zf — APPROVED** [review]
Ok this time I think we're done 😆 
Doing manual sweep and I'll merge when I'm done with that.

**Review by borisrizov-zf — CHANGES_REQUESTED** [review]

**Comment by borisrizov-zf** [comment]
@andreibogus @aleksandra-bel some minor findings.

**Review by borisrizov-zf — APPROVED** [review]

**Pull request merged**